src/aleph/netty.clj

```diff
@@ -1176,27 +1255,44 @@
       daemon?
       (fn [group target name stack-size] (FastThreadLocalThread. group target name stack-size)))))
 
+(defn ^:no-doc enumerating-thread-factory
+  "Sets the ThreadFactory used by Netty EventLoopGroups.
+
+   Sets the ClassLoader to the Clojure DynamicClassLoader for any clj
+   code that could be loaded by Netty."
+  ^ThreadFactory
+  [prefix daemon?]
+  (let [num-threads (atom 0)]
+    (e/thread-factory
+      #(str prefix "-" (swap! num-threads inc))
+      (deliver (promise) nil)
+      nil
+      daemon?
+      (fn [group target name stack-size]
+        (doto (FastThreadLocalThread. group target name stack-size)
+              (.setContextClassLoader (DynamicClassLoader.)))))))
```

Comment:
Why was this necessary for HTTP2?

Reply:
Don't recall it off the top of my head, but according to a35226a0, there was an issue with byte-streams.graph.Type being loaded by two different classloaders.